bolt-modules/boltlib/lib/puppet/functions/puppetdb_fact.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+# Returns a hash of certname to facts objects for each matched Target.
+# This functions accepts a TargetSpec, and returns a hash of target certnames
+# and their corresponding facts hash.
+#
+# * If a node is not found in PuppetDB, it's included in the returned hash with empty facts hash.
+# * Otherwise the node is included in the hash with a value that is a hash of it's facts.
+#
+Puppet::Functions.create_function(:puppetdb_fact) do
+  dispatch :puppetdb_fact do
+    param 'Boltlib::TargetSpec', :targets
```

Comment:
We should require this to be an array of strings (certnames), and handle turning TargetSpec into an array of strings in the wrapper plan.

Reply:
Currently you accept a TargetSpec, but then expect it to be a list of certnames in the executor.

Reply:
Yep yep. Updated!